library/riscv-aia/src/peripheral/aplic.rs

```diff
@@ -521,6 +524,416 @@ mod tests {
+    }
+
+    #[test]
+    #[should_panic(expected = "High base PPN out of range: 0..=0xFFF")]
+    fn test_machine_msi_addr_cfg_h_high_base_ppn_out_of_range() {
+        let config = MachineMsiAddrCfgH(0x0000_0000);
+        config.set_high_base_ppn(0x1000);
+    }
+}
 
-    // TODO unit tests for functions of DomainConfig and other structures.
+#[cfg(test)]
+mod supervisor_msi_addr_cfg_h_tests {
+    use super::SupervisorMsiAddrCfgH;
+
+    #[test]
+    fn test_supervisor_msi_addr_cfg_h_low_hart_index_shift() {
+        let config = SupervisorMsiAddrCfgH(0x0000_0000);
+        let config = config.set_low_hart_index_shift(0);
+        assert_eq!(config.low_hart_index_shift(), 0);
+
+        let config = config.set_low_hart_index_shift(7);
+        assert_eq!(config.low_hart_index_shift(), 7);
+    }
+
+    #[test]
+    #[should_panic(expected = "Low hart index shift out of range: 0..=7")]
+    fn test_supervisor_msi_addr_cfg_h_low_hart_index_shift_out_of_range() {
+        let config = SupervisorMsiAddrCfgH(0x0000_0000);
+        config.set_low_hart_index_shift(8);
+    }
+
+    #[test]
+    fn test_supervisor_msi_addr_cfg_h_high_base_ppn() {
+        let config = SupervisorMsiAddrCfgH(0x0000_0000);
+        let config = config.set_high_base_ppn(0);
+        assert_eq!(config.high_base_ppn(), 0);
+
+        let config = config.set_high_base_ppn(0xFFF);
+        assert_eq!(config.high_base_ppn(), 0xFFF);
+    }
+
+    #[test]
+    #[should_panic(expected = "High base PPN out of range: 0..=0xFFF")]
+    fn test_supervisor_msi_addr_cfg_h_high_base_ppn_out_of_range() {
+        let config = SupervisorMsiAddrCfgH(0x0000_0000);
+        config.set_high_base_ppn(0x1000);
+    }
+}
+
+#[cfg(test)]
+mod set_int_pending_tests {
+    use super::SetIntPending;
+
+    #[test]
+    fn test_set_int_pending() {
+        let pending = SetIntPending(0x0000_0000);
+        let pending = pending.set_int_pending(0x1234_5678);
+        assert_eq!(pending.int_pending(), 0x1234_5678);
+    }
+}
+
+#[cfg(test)]
+mod clear_int_pending_tests {
+    use super::ClearIntPending;
+
+    #[test]
+    fn test_clear_int_pending() {
+        let pending = ClearIntPending(0x0000_0000);
+        let pending = pending.clear_int_pending(0x1234_5678);
+        assert_eq!(pending.int_pending(), 0x1234_5678);
+    }
+}
+
+#[cfg(test)]
+mod set_int_enable_tests {
+    use super::SetIntEnable;
+
+    #[test]
+    fn test_set_int_enable() {
+        let enable = SetIntEnable(0x0000_0000);
+        let enable = enable.set_int_enable(0x1234_5678);
+        assert_eq!(enable.int_enable(), 0x1234_5678);
+    }
+}
+
+#[cfg(test)]
+mod clear_int_enable_tests {
+    use super::ClearIntEnable;
+
+    #[test]
+    fn test_clear_int_enable() {
+        let enable = ClearIntEnable(0x0000_0000);
+        let enable = enable.clear_int_enable(0x1234_5678);
+        assert_eq!(enable.int_enable(), 0x1234_5678);
+    }
+}
+
+#[cfg(test)]
+mod generate_msi_tests {
+    use super::GenerateMSI;
```

Comment:
```suggestion
    use super::GenerateMsi;
```

Rename this struct into `GenerateMsi`.